WooCommerce/Classes/ViewRelated/Coupons/CouponDetails/CouponDetailsViewModel.swift

```diff
@@ -0,0 +1,75 @@
+import Foundation
+import Yosemite
+
+/// View model for `CouponDetails` view
+///
+final class CouponDetailsViewModel: ObservableObject {
+    /// Code of the coupon
+    ///
+    @Published private(set) var couponCode: String = ""
+
+    /// Description of the coupon
+    ///
+    @Published private(set) var description: String = ""
+
+    /// Amount of the coupon
+    ///
+    @Published private(set) var amount: String = ""
+
+    /// Product limit for the coupon to be applied to
+    ///
+    @Published private(set) var applyTo: String = ""
```

Comment:
nit: it's a bit hard to tell what this variable means until I read the comment, maybe we can mention products here? like:
```suggestion
    @Published private(set) var productsAppliedTo: String = ""
```

Reply:
Renamed in 5803698fc52345a58998ac2fbf6af96abc6c96ab 👍